File: rust/src/src/base/i18n/i18n_resolver.rs
use std::collections::HashMap;

use color_eyre::eyre::Result;

use super::i18n_service;

pub async fn n_lang(
  lang_code: String,
  route_path: Option<String>,
  code: String,
  args: Option<HashMap<String, String>>,
) -> Result<String> {
  
  let res = i18n_service::n_lang(
    lang_code,
    route_path,
    code,
    args,
  ).await?;
  
  Ok(res)
}

File: deno/src/base/i18n/i18n.resolver.ts
import {
  setNotVerifyToken,
} from "/lib/context.ts";

/**
 * 国际化
 * @param langCode 语言编码, 例如: zh-CN
 * @param routePath 页面路由, 例如: /user
 * @param code 国际化编码
 */
export async function n(
  langCode: string,
  routePath: string | null,
  code: string,
) {
  
  const {
    nLang,
  } = await import("./i18n.ts");
  
  setNotVerifyToken(true);
  
  return await nLang(langCode, routePath, code);
}
